tests/publiccloud/check_registercloudguest.pm

```diff
@@ -122,8 +122,8 @@ sub run {
 
 sub check_instance_registered {
     my ($instance) = @_;
-    if ($instance->ssh_script_output(cmd => 'zypper -x lr | grep "<repo\s" | wc -l', timeout => 300) == 0) {
-        record_info('zypper lr', $instance->ssh_script_output(cmd => 'zypper lr'));
+    if ($instance->ssh_script_output(cmd => 'LANG=C zypper -t lr | awk "/^\s?[[:digit:]]+/{c++} END {print c}"', timeout => 300) == 0) {
+        record_info('zypper lr', $instance->ssh_script_output(cmd => 'zypper -t lr ||:'));
```

Comment:
and what `||:` suppose to do ? I tried to run `zypper lr` with and without it and it looks similar

Reply:
`zypper lr` returns exit code `6` if there isnt repo, so script osutpuit will fail , so `||:` forces in that cases `true` so return code stays `0`